python/ray/tests/test_failure.py

```diff
@@ -322,6 +322,92 @@ def foo():
         assert isinstance(ex, RayTaskError)
 
 
+def test_base_exception_task(ray_start_regular):
+    @ray.remote
+    def task():
+        raise BaseException("abc")
+
+    with pytest.raises(ray.exceptions.WorkerCrashedError):
+        ray.get(task.remote())
+
+
+def test_base_exception_actor(ray_start_regular):
+    @ray.remote
+    class Actor:
+        def f(self):
+            raise BaseException("abc")
+
+    with pytest.raises(ray.exceptions.ActorDiedError):
+        a = Actor.remote()
+        ray.get(a.f.remote())
+
+
+@pytest.mark.skipif(
+    sys.version_info < (3, 11),
+    reason="BaseExceptionGroup is only available in Python 3.11+",
+)
+def test_base_exception_group_task(ray_start_regular):
+    @ray.remote
+    def task():
+        raise BaseExceptionGroup("abc", [BaseException("def")])  # noqa: F821
+
+    with pytest.raises(ray.exceptions.WorkerCrashedError):
+        ray.get(task.remote())
+
+
+@pytest.mark.skipif(
+    sys.version_info < (3, 11),
+    reason="BaseExceptionGroup is only available in Python 3.11+",
+)
+def test_base_exception_group_actor(ray_start_regular):
+    @ray.remote
+    class Actor:
+        def f(self):
+            raise BaseExceptionGroup("abc", [BaseException("def")])  # noqa: F821
+
+    with pytest.raises(ray.exceptions.ActorDiedError):
+        a = Actor.remote()
+        ray.get(a.f.remote())
+
+
+@pytest.mark.skipif(
+    sys.version_info < (3, 11),
+    reason="ExceptionGroup is only available in Python 3.11+",
+)
+def test_exception_group(ray_start_regular):
```

Comment:
can you also do a test for the new `except*` syntax?

Reply:
Done.

I moved `ExceptionGroup` related tests to `python/ray/tests/future_python_syntax/test_exceptiongroup_py311.py` and skip them with `collect_ignore_glob` because the `except*` syntax will cause syntax error in Python < 3.11, so `pytest.mark.skipif` cannot be used.

Here is one behavior that I think need to discuss with you. For the

```python
except* RayTaskError as ex:
```

case, `ex.exceptions` does not contain `RayTaskError`. I don't know whether users may be confused why they catched `RayTaskError` but there is no `RayTaskError` in the `ExceptionGroup`. But including `RayTaskError` into `ex.exceptions` is also weird because it will result in self-referencing.

Reply:
I think it's ok to not include RayTaskError into the group. The expected behavior:

```
try:
    ray.get(raises)
except RayTaskError:  # can catch
    pass
except ExceptionGroup:  # can catch
    pass
except* MyUserError:  # can catch
    pass
except* RayTaskError:  # can NOT catch
    pass
```

Reply:
I think the last one is not technically doable. If an Exception can be caught by the `except` statement, it can also be caught by the `except*` statement.

For example:

```python
try:
    raise ValueError('This is the error message.')
except* ValueError as ex:
    print(repr(ex))
```

The output would be 

```
ExceptionGroup('', (ValueError('This is the error message.'),))
```

Reply:
understood. then we don't need to test that case.